app/db.py

```diff
@@ -13,72 +22,116 @@ def connect():
-        port=os.environ.get('DB_PORT'))
+    engine = create_engine("postgresql://{0}@{1}:5432/{2}".format(
+        os.environ.get('DB_USER'),
+        os.environ.get('DB_HOST'),
+        db))
+    Session.configure(bind=engine)
+    Base.metadata.create_all(engine)
+    engine.connect()
+
+
+@contextmanager
+def session_scope():
+    """ generator for handling sessions for database interactions"""
+    session = Session()
+    try:
+        yield session
+        session.commit()
+    except Exception as e:
+        print  e
+        session.rollback()
+    finally:
+        session.close()
+
 
+class OHLCV(Base):
+    __tablename__ = 'ohlcv'
 
-conn = connect()
+    id = Column(Integer, primary_key=True)
+    date = Column(Date)
+    symbol = Column(String)
+    open = Column(String)
+    high = Column(String)
+    low = Column(String)
+    close = Column(String)
+    volume = Column(String)
 
 
 def symbol_exists(symbol):
-    """ check if the symbol exists in the database """
-    with conn.cursor() as cursor:
-        cursor.execute("SELECT exists (SELECT 1 FROM ohlcv WHERE symbol = %s LIMIT 1)", (symbol,))
-        symbol, = cursor.fetchone()
-        return symbol
+    """ check if the stock symbol exists in the database """
+    return session.query(exists().where(OHLCV.symbol==symbol))[0]
 
 
 def get_recent_ohlvc(symbol):
     """ get the last 10 days of data for a stock symbol """
-    with conn.cursor() as cursor:
-        cursor.execute("SELECT * FROM ohlcv WHERE symbol = %s ORDER BY date DESC LIMIT 10", (symbol,))
-        recent_ohlvc_data = cursor.fetchall()
-        lastTenDaysData = []
-        for day in recent_ohlvc_data: 
-            lastTenDaysData.append([str(value) for value in day])
-        # FIXME should transform data here to something like:
-        # {'symbol': lastTenDaysData[0], 'open': lastTenDaysData[1]...etc}
-        # then update route and view where this is rendered
-        return lastTenDaysData
-
-
-def get_start_date(symbol):
-    """ get the most recent date that we have daily stock data """
-    with conn.cursor() as cursor:
-        cursor.execute("SELECT MAX(date) FROM ohlcv WHERE symbol = %s", (symbol,))
-        start_date, = cursor.fetchone() # returns none if there are no records
+    lastTenDays = []
+    with session_scope() as session:
+        data = session.query(OHLCV). \
+            filter_by(symbol=symbol). \
+            order_by(OHLCV.date.desc()). \
+            limit(10)
+
+    for day in data:
+        dayDict = dict(
+            date=day.date,
+            symbol=day.symbol,
+            open=day.open,
+            high=day.high,
+            low=day.low,
+            close=day.close,
+            volume=day.volume
+            )
+        lastTenDays.append(dayDict)
+
+    return lastTenDays
 
-    if start_date:
-        start_date += timedelta(days=1)
 
+def get_recent_data_date(symbol):
+    """ get the most recent date that we have daily stock data """
+    start_date = ''
```

Comment:
Why is `start_date` an empty string?  As far as I can tell, the return value of this functions is only used in operations with dates and timedeltas.  For example, line 133 `delta_date = recent_data_date - last_data_day` should cause an error if `recent_data_date` is an empty string.